testing/internal/e2e/target/target_test.go

```diff
@@ -0,0 +1,280 @@
+	t.Logf("Created Org Id: %s", newOrg.Id)
+
+	// Create a project
+	output, err = e2e.RunCommand(
+		"boundary", "scopes", "create",
+		"-scope-id", newOrg.Id,
+		"-name", "e2e Automated Test Project",
+		"-format", "json",
+	)
+	require.NoError(t, err, string(output.Stderr))
+	var newProjectResult scopes.ScopeCreateResult
+	err = json.Unmarshal(output.Stdout, &newProjectResult)
+	require.NoError(t, err)
+	newProject := newProjectResult.Item
+	t.Cleanup(func() {
+		output, err := e2e.RunCommand("boundary", "scopes", "delete", "-id", newProject.Id)
+		require.NoError(t, err, string(output.Stderr))
+	})
+	t.Logf("Created Project Id: %s", newProject.Id)
+
+	// Create a host catalog
+	output, err = e2e.RunCommand(
+		"boundary", "host-catalogs", "create", "static",
+		"-scope-id", newProject.Id,
+		"-name", "e2e Automated Test Host Catalog",
+		"-format", "json",
+	)
+	require.NoError(t, err, string(output.Stderr))
+	var newHostCatalogResult hostcatalogs.HostCatalogCreateResult
+	err = json.Unmarshal(output.Stdout, &newHostCatalogResult)
+	require.NoError(t, err)
+	newHostCatalog := newHostCatalogResult.Item
+	t.Cleanup(func() {
+		output, err := e2e.RunCommand("boundary", "host-catalogs", "delete", "-id", newHostCatalog.Id)
+		require.NoError(t, err, string(output.Stderr))
+	})
+	t.Logf("Created Host Catalog: %s", newHostCatalog.Id)
+
+	// Create a host set and add to catalog
+	output, err = e2e.RunCommand(
+		"boundary", "host-sets", "create", "static",
+		"-host-catalog-id", newHostCatalog.Id,
+		"-name", "e2e Automated Test Host Set",
+		"-format", "json",
+	)
+	require.NoError(t, err, string(output.Stderr))
+	var newHostSetResult hostsets.HostSetCreateResult
+	err = json.Unmarshal(output.Stdout, &newHostSetResult)
+	require.NoError(t, err)
+	newHostSet := newHostSetResult.Item
+	t.Cleanup(func() {
+		output, err := e2e.RunCommand("boundary", "host-sets", "delete", "-id", newHostSet.Id)
+		require.NoError(t, err, string(output.Stderr))
+	})
+	t.Logf("Created Host Set: %s", newHostSet.Id)
+
+	// Create a host
+	output, err = e2e.RunCommand(
+		"boundary", "hosts", "create", "static",
+		"-host-catalog-id", newHostCatalog.Id,
+		"-name", c.TargetIp,
+		"-address", c.TargetIp,
+		"-format", "json",
+	)
+	require.NoError(t, err, string(output.Stderr))
+	var newHostResult hosts.HostCreateResult
+	err = json.Unmarshal(output.Stdout, &newHostResult)
+	require.NoError(t, err)
+	newHost := newHostResult.Item
+	t.Cleanup(func() {
+		output, err := e2e.RunCommand("boundary", "hosts", "delete", "-id", newHost.Id)
+		require.NoError(t, err, string(output.Stderr))
+	})
+	t.Logf("Created Host: %s", newHost.Id)
+
+	// Add host to host set
+	output, err = e2e.RunCommand(
+		"boundary", "host-sets", "add-hosts",
+		"-id", newHostSet.Id,
+		"-host", newHost.Id,
+	)
+	require.NoError(t, err, string(output.Stderr))
+
+	// Create a target
+	output, err = e2e.RunCommand(
+		"boundary", "targets", "create", "tcp",
+		"-scope-id", newProject.Id,
+		"-default-port", "22",
```

Comment:
You might want to make this configurable via an env var. I was able to run these tests against a docker setup but my ssh server was running on `2222` so I needed to change this port. (The test still failed but that was due to a the `hostname -i` command returning a docker ip instead of the address from the host, which is a separate problem that this normally will not encounter).

Reply:
Should we think of an alternative to checking `hostname -i` so that it would work in this docker environment?

Reply:
Keeping the `hostname` check as is for now. Will revisit in the future.